torchdata/nodes/samplers/multi_node_weighted_sampler.py

```diff
@@ -107,12 +111,15 @@ def reset(self, initial_state: Optional[Dict[str, Any]] = None):
         super().reset(initial_state)
 
         if initial_state is not None:
+            self._initial_state = initial_state
```

Comment:
When will self._initial_state be reset? I'd prefer a solution where we don't need to hold on to this